File: Observer/PHP/MaxObserver.php
<?php

require_once "IObserver.php";

class MaxObserver implements IObserver {
    private int $maxCPU = 0, 
                $maxRAM = 0;

    public function update(int $CPU, int $RAM) {
        $this->maxCPU = max($this->maxCPU, $CPU);
        $this->maxRAM = max($this->maxRAM, $RAM);

        echo "Current Max CPU: {$this->maxCPU}\nCurrent Max RAM: {$this->maxRAM}\n";
    }
}

File: Observer/Ruby/MaxObserver.rb
require_relative "IObserver"

class MaxObserver < IObserver
    def initialize
        @maxCPU = 0
        @maxRAM = 0
    end

    def update(cpu, ram)
        @maxCPU = [@maxCPU, cpu].max
        @maxRAM = [@maxRAM, ram].max

        puts "Current Max CPU: #{@maxCPU}\nCurrent Max RAM: #{@maxRAM}"
    end
end